packages/ember/src/components/form/phone-field.gts

```diff
@@ -0,0 +1,120 @@
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { runTask } from 'ember-lifeline';
+
+import TextField from './text-field.gts';
+import onInsert from '../../modifiers/did-insert.ts';
+
+import type { Optional } from '../../types.d.ts';
+
+export default class PhoneField extends TextField {
+  @tracked
+  inputElement: Optional<HTMLInputElement> = null;
+
+  @action
+  onInsert(element: HTMLElement) {
+    this.inputElement = element as HTMLInputElement;
+  }
+
+  @action
+  change(evt: Event) {
+    const inputEvent = evt as InputEvent;
+    const target = evt.target as HTMLInputElement;
+    const newValue = target?.value;
+    const currentValue = this.displayValue;
+    const cursorPosition = target?.selectionStart ?? -1;
+    const isBackspace = inputEvent.inputType === 'deleteContentBackward';
+    const isDelete = inputEvent.inputType === 'deleteContentForward';
+
+    let unformattedValue = newValue.replace(/\D/g, '');
```

Comment:
With so much extra logic to create a better user experience, can tests be added for various scenarios?